mobile/native/src/ln_dlc/mod.rs

```diff
@@ -560,6 +504,128 @@ fn keep_wallet_balance_and_history_up_to_date(node: &Node) -> Result<()> {
     Ok(())
 }
 
+fn derive_trades_from_filled_orders() -> Result<Vec<WalletHistoryItem>> {
+    let mut trades = vec![];
+    let orders =
+        crate::db::get_filled_orders().context("Failed to get filled orders; skipping update")?;
+
+    match orders.as_slice() {
+        [first, tail @ ..] => {
+            // The first filled order must be an outbound "payment", since coins need to leave the
+            // Lightning wallet to open the first DLC channel.
+            let flow = PaymentFlow::Outbound;
+            let amount_sats = first
+                .trader_margin()
+                .expect("Filled order to have a margin");
+
+            trades.push(WalletHistoryItem {
+                flow,
+                amount_sats,
+                timestamp: first.creation_timestamp.unix_timestamp() as u64,
+                status: Status::Confirmed, // TODO: Support other order/trade statuses
+                wallet_type: WalletHistoryItemType::Trade {
+                    order_id: first.id.to_string(),
+                },
+            });
+
+            let mut total_contracts = match first.direction {
+                Direction::Long => first.quantity,
+                Direction::Short => -first.quantity,
+            };
+            let mut previous_order = first;
+            for order in tail {
+                use trade::Direction::*;
+                let new_contracts = match order.direction {
+                    Long => order.quantity,
+                    Short => -order.quantity,
+                };
+                let updated_total_contracts = total_contracts + new_contracts;
+
+                // Closing the position.
+                if updated_total_contracts == 0.0 {
+                    let open_order = previous_order;
+                    let trader_margin = open_order
+                        .trader_margin()
+                        .expect("Filled order to have a margin");
```

Comment:
Should we handle this error ? Otherwise we panic and 💥

Reply:
Hmm, this one didn't change and we just know that a filled order must have a margin. It sucks to not use the type system for this, but we chose to create an `orders` table with nullable values.

Reply:
Got ya